logserver.c

```diff
@@ -678,15 +936,69 @@ void pv_logserver_stop(void)
 	return logserver_stop();
 }
 
-void pv_logserver_close()
+static void logserver_close(int sockd, const char *name)
 {
+	if (sockd < 0)
+		return;
+
 	char path[PATH_MAX];
+	pv_paths_pv_file(path, PATH_MAX, name);
+	pv_log(DEBUG, "closing %s with fd %d\n", path, sockd);
+	close(sockd);
+	unlink(path);
+}
 
-	if (logserver_g.sock_fd >= 0) {
-		pv_paths_pv_file(path, PATH_MAX, LOGCTRL_FNAME);
-		pv_log(DEBUG, "closing %s with fd %d", path,
-		       logserver_g.sock_fd);
-		close(logserver_g.sock_fd);
-		unlink(path);
-	}
+void pv_logserver_close()
+{
+	if (logserver_g.logsock >= 0)
+		logserver_close(logserver_g.logsock, LOGCTRL_FNAME);
+	if (logserver_g.fdsock >= 0)
+		logserver_close(logserver_g.fdsock, LOGFD_FNAME);
+}
+
+int pv_logserver_subscribe_fd(int fd, const char *platform, const char *src)
+{
+	if (!logserver_g.out.sfile)
+		return 0;
+
+	char plat_buf[LOGSERVER_MAX_HEADER_LEN] = { 0 };
+	char src_buf[LOGSERVER_MAX_HEADER_LEN] = { 0 };
+
+	strncpy(plat_buf, platform, LOGSERVER_MAX_HEADER_LEN - 1);
+	strncpy(src_buf, src, LOGSERVER_MAX_HEADER_LEN - 1);
+
+	struct iovec iov[2];
+	iov[0] = (struct iovec){ .iov_base = plat_buf,
+				 .iov_len = LOGSERVER_MAX_HEADER_LEN };
+	iov[1] = (struct iovec){ .iov_base = src_buf,
+				 .iov_len = LOGSERVER_MAX_HEADER_LEN };
+
+	union {
+		char buf[CMSG_SPACE(sizeof(int))];
+		struct cmsghdr align;
+	} ctrl;
+
+	struct msghdr msg = { .msg_name = NULL,
```

Comment:
Not very strong on this, but could we get a send message function for sending messages in general? do you think it would be useful some day?

Reply:
It could be! but I think we can do something like LXC af_unix lib who includes functions to connect, listen, send/receive fd, etc.
https://gitlab.com/pantacor/lxc/-/blob/stable-3.0/src/lxc/af_unix.h